lib/LoginStrategy.ts

```diff
@@ -424,6 +486,29 @@ export class LoginStrategy extends BaseStrategy {
     return true;
   }
 
+  isSiweMessagePayload(
+    payload: unknown
+  ): payload is Partial<SiweMessagePayload> {
+    if (!payload) return false;
+    if (typeof payload !== 'object') return false;
+    if (
+      !hasIn(payload, 'domain') ||
+      !hasIn(payload, 'nonce') ||
+      !hasIn(payload, 'uri') ||
+      !hasIn(payload, 'address')
+    ) {
+      return false;
+    }
+    if (
+      typeof payload['domain'] !== 'string' ||
+      typeof payload['nonce'] !== 'string' ||
+      typeof payload['uri'] !== 'string'
+    ) {
+      return false;
+    }
```

Comment:
I think the `hasIn` checks are redundant here because if the key isn't on the object then the type will be `undefined` which will be caught by the type checks.
I also think that by casting the property to the type we want, we can have compilation errors in the future if we upgrade `siwe` package and the `SiweMessage` class changes.
```suggestion
    const castedPayload = payload as SiweMessage;
    if (
      typeof castedPayload.domain !== 'string' ||
      typeof castedPayload.nonce !== 'string' ||
      typeof castedPayload.uri !== 'string' ||
      typeof castedPayload.address !== 'string'
    ) {
      return false;
    }
```